docs/get-started/cardanosharp-wallet.md

```diff
@@ -92,140 +91,226 @@ Address baseAddr = addressService.GetAddress(
     AddressType.Base);
 ```
 
-### NetworkType
+If you already have an address.
 
-```csharp
-namespace CardanoSharp.Wallet.Enums
-{
-    public enum NetworkType
-    {
-        Testnet,
-        Mainnet
-    }
-}
+```cs
+Address baseAddr = new Address("addr_test1qz2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzer3jcu5d8ps7zex2k2xt3uqxgjqnnj83ws8lhrn648jjxtwq2ytjqp");
 ```
 
-### AddressType
+## Fluent Key Derivation
 
-```csharp
-namespace CardanoSharp.Wallet.Enums
-{
-    public enum AddressType
-    {
-        Base,
-        Enterprise,
-        Reward
-    }
-}
+A fluent API helps navigate the derivation paths.
+
+```cs
+// Add using
+using CardanoSharp.Wallet.Extensions.Models;
+
+// Restore a Mnemonic
+var mnemonic = new MnemonicService().Restore(words);
+
+// Fluent derivation API
+var derivation = mnemonic
+    .GetMasterNode("password")      // IMasterNodeDerivation
+    .Derive(PurposeType.Shelley)    // IPurposeNodeDerivation
+    .Derive(CoinType.Ada)           // ICoinNodeDerivation
+    .Derive(0)                      // IAccountNodeDerivation
+    .Derive(RoleType.ExternalChain) // IRoleNodeDerivation
+    //or .Derive(RoleType.Staking) 
+    .Derive(0);                     // IIndexNodeDerivation
+
+PrivateKey privateKey = derivation.PrivateKey;
+PublicKey publicKey = derivation.PublicKey;
 ```
 
 ## Build and Sign Transactions
 
-This is just an example of how to start. You will need to Calculate Fees, compare with Protocol Parameters and re-serialize.
+CardanoSharp.Wallet requires input from the chain in order to build transactions. Lets assume we have gathered the following information.
```

Comment:
another minor, and sorry, not really important but "let's assume" sounds more correct also in 4 other lines